app/src/main/java/com/github/swent/echo/compose/map/Map.kt

```diff
@@ -1,59 +1,62 @@
 import androidx.compose.ui.viewinterop.AndroidView
-import org.osmdroid.config.Configuration
-import org.osmdroid.tileprovider.MapTileProviderBasic
-import org.osmdroid.tileprovider.tilesource.ITileSource
-import org.osmdroid.tileprovider.tilesource.TileSourceFactory
-import org.osmdroid.util.GeoPoint
-import org.osmdroid.views.MapView
-
-val LAUSANNE_GEO_POINT: GeoPoint = GeoPoint(46.5197, 6.6323)
-const val ZOOM_DEFAULT = 15.0
-
-fun configureOsmdroid(context: Context) {
-    Configuration.getInstance().apply {
-        userAgentValue = context.packageName
-        osmdroidBasePath = context.cacheDir
-    }
-}
-
-fun createMapView(context: Context, tileSource: ITileSource): MapView =
-    MapView(context, MapTileProviderBasic(context)).apply {
-        setTileSource(tileSource)
-        // setOnClickListener { ... }
-        controller.setZoom(ZOOM_DEFAULT)
-        controller.setCenter(LAUSANNE_GEO_POINT)
-        clipToOutline = true
-    }
+import com.github.swent.echo.data.model.Event
 
-fun updateMapView(view: MapView, newCenter: GeoPoint) {
-    view.apply { controller.setCenter(newCenter) }
+@Composable
+fun <T : View> EchoAndroidView(
+    modifier: Modifier = Modifier,
+    factory: (Context) -> T,
+    update: (T, List<Event>, (Event) -> Unit) -> Unit,
+    events: MutableState<List<Event>>,
+    callback: (Event) -> Unit = {}
+) {
+    AndroidView(
+        modifier = modifier,
+        factory = factory,
+        update = { update(it, events.value, callback) }
+    )
 }
 
-@Preview
+/**
+ * This composable will draw a map that displays events.
+ *
+ * @param context Needed to initialize the [MapViewProvider][IMapViewProvider]. Can be the activity
+ *   itself if this function is called from an activity, or the `current` field of a
+ *   [LocalContext][androidx.compose.ui.platform.LocalContext] if it is called from another
+ *   composable.
+ * @param events The list of events to be displayed by the map. Whenever this list changes, the
+ *   event markers in the map will be automatically updated to match the list. Note that this
+ *   parameter is a [MutableState] containing a `List`, so the correct way to instantiate it is the
+ *   following: `val myEvents = remember { mutableStateOf(emptyList<Event>()) }`. Do NOT use the
+ *   `by` keyword. You can then pass it as an argument to this composable. If you want to change the
+ *   value stored in the [MutableState] to trigger a redraw of the markers, you can write
+ *   `myEvents.value = newEventList`.
+ * @param callback The function to be called when a marker on the map is clicked on. The [Event]
+ *   corresponding to the marker will be passed as an argument to this function.
+ */
 @Composable
 fun MapDrawer(
     modifier: Modifier = Modifier,
-    mapViewFactory: (Context) -> MapView = { createMapView(it, TileSourceFactory.MAPNIK) },
-    update: (MapView) -> Unit = { updateMapView(it, LAUSANNE_GEO_POINT) }
+    context: Context,
+    events: MutableState<List<Event>>,
```

Comment:
I saw that the `events` param is expected to be a `MutableState<...>`. I was wondering why not simply take a `List<Event>` here? So that outside of the `MapDrawer`, it is possible to instantiate the list as `val myEvents by remember { mutableStateOf(emptyList<Event>()) }`. And then simply pass the `myEvents` to the composable, it would prevent the need to call the `.value` when updating the list outside.

I haven't done the test myself, would it prevent the `MapDrawer` from recomposing ?

Reply:
No, you''re right. I had my doubts about whether the `AndroidView` would be properly recomposed without the `MutableState`, but it seems to work just fine. Thanks!